cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/SearchController.java

```diff
@@ -0,0 +1,36 @@
+package edu.asu.diging.cord19.explorer.web;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.diging.cord19.explorer.core.mongo.impl.SearchProviderRegistry;
+import edu.asu.diging.cord19.explorer.web.model.SearchType;
+
+@Controller
+public class SearchController {
+
+    @Autowired
+    private SearchProviderRegistry searchProviderRegistry;
+
+    @RequestMapping(value = "/{searchType}/search")
+    public String search(@PathVariable("searchType") String searchType, @RequestParam("search") String query,
+            Model model, @PageableDefault(size = 20) Pageable pageable) {
+
+        searchType = searchType.toUpperCase();
+        long count = searchProviderRegistry.getProvider(SearchType.valueOf(searchType)).getTotalResults(query);
```

Comment:
`valueOf` throws an `IllegalArgumentException` when you pass a value that doesn't exist. You should handle that here.